### PR TITLE
Address bar polish

### PR DESCRIPTION
**Pull request by venkatn93**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1be303e3ed8cd2f812aa7e8fe37652321652508d available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 564549117031124cc13a0cceb6fe71f86cb164f2 available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 894a24531834e875acb63e000ccb061d13777897 available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: afc2db6c6a77a874df4f681faaf724861d90988d available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

File: libraries/networking/src/AddressManager.h [hunk 1]
```diff
@@ -62,6 +62,9 @@ class AddressManager : public QObject, public Dependency {
 
     void loadSettings(const QString& lookupString = QString());
 
+    bool getBackState() { return (!_backStack.isEmpty()); }
```
Comment (jherico) [inline libraries/networking/src/AddressManager.h]: I'd suggest just exposing the back and forward as read only properties...

i.e. `const QStack<QUrl>& getBackStack() const { return _backStack; }`

File: interface/src/ui/AddressBarDialog.h [hunk 2]
```diff
@@ -18,18 +18,31 @@ class AddressBarDialog : public OffscreenQmlDialog
...
+
+signals:
+    void backEnabledChanged();
+    void forwardEnabledChanged();
 
 protected:
     void displayAddressOfflineMessage();
     void displayAddressNotFoundMessage();
     void hide();
+    void changeBackAlpha(bool isPossible);
```
Comment (jherico) [inline interface/src/ui/AddressBarDialog.h]: the naming of these functions doesn't make sense.

File: interface/src/ui/AddressBarDialog.cpp [hunk 3]
```diff
@@ -22,12 +22,34 @@ AddressBarDialog::AddressBarDialog(QQuickItem* parent) : OffscreenQmlDialog(pare
     connect(addressManager.data(), &AddressManager::lookupResultIsOffline, this, &AddressBarDialog::displayAddressOfflineMessage);
     connect(addressManager.data(), &AddressManager::lookupResultIsNotFound, this, &AddressBarDialog::displayAddressNotFoundMessage);
     connect(addressManager.data(), &AddressManager::lookupResultsFinished, this, &AddressBarDialog::hide);
+    connect(addressManager.data(), &AddressManager::goBackPossible, this, &AddressBarDialog::changeBackAlpha);
```
Comment (jherico) [inline interface/src/ui/AddressBarDialog.cpp]: in order to reduce the surface area of the class I tend to use lambdas in connections when I can...  

```
connect(addressManager.data(), &AddressManager::goBackPossible, [this](bool isPossible){ 
    if (isPossible != _backEnabled) {
        _backEnabled = isPossible;
        emit backEnabledChanged();
    }
});
```

This removes the need for having additional functions just to handle a given signal.

File: interface/src/ui/AddressBarDialog.cpp [hunk 4]
```diff
@@ -22,12 +22,34 @@ AddressBarDialog::AddressBarDialog(QQuickItem* parent) : OffscreenQmlDialog(pare
...
+    _backEnabled = DependencyManager::get<AddressManager>()->getBackState();
+    _forwardEnabled = DependencyManager::get<AddressManager>()->getForwardState();
 }
 
 void AddressBarDialog::hide() {
     ((QQuickItem*)parent())->setEnabled(false);
 }
 
+void AddressBarDialog::changeBackAlpha(bool isPossible) {
+    if (isPossible) qDebug() << "Back is possible";
```
Comment (jherico) [inline interface/src/ui/AddressBarDialog.cpp]: remove debugging code if it's not needed anymore.  

Also, our coding standards mandate the use of braces in conditionals...
```
if (x) { 
  a();
} else {
  b();
}
```

File: interface/resources/qml/AddressBarDialog.qml [hunk 5]
```diff
@@ -88,6 +101,14 @@ DialogContainer {
                         addressBarDialog.loadForward()
                     }
                 }
+
+                /*
```
Comment (jherico) [inline interface/resources/qml/AddressBarDialog.qml]: remove dead code

File: interface/resources/qml/AddressBarDialog.qml [hunk 6]
```diff
@@ -66,12 +70,21 @@ DialogContainer {
                         addressBarDialog.loadBack()
                     }
                 }
+
+                /*
```
Comment (jherico) [inline interface/resources/qml/AddressBarDialog.qml]: remove dead code

File: interface/resources/qml/AddressBarDialog.qml [hunk 7]
```diff
@@ -30,12 +30,17 @@ DialogContainer {
...
     property int maximumY: parent ? parent.height - height : 0
 
+
     AddressBarDialog {
         id: addressBarDialog
 
         implicitWidth: backgroundImage.width
         implicitHeight: backgroundImage.height
 
+        onParentChanged: {
```
Comment (jherico) [inline interface/resources/qml/AddressBarDialog.qml]: remove debugging code

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1494aa4faa5c9e30e1453bb0e209d90432c1bc22 available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 054e12a3151a5c8c636acfd36cb415cd31915f1c available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 03c10d15e6080ce0800b47a1380205a9ccc82842 available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 482f12be8cf22c3a65be472cbbfe2629ca8cfab0 available at: https://gist.github.com/37aac90365d81922906d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5163.dmg

**Pull request merged**